test/problematic_tests/save_genome_test.py

```diff
@@ -142,29 +150,46 @@ def start_test(self):
         testname = inspect.stack()[1][3]
         print(('\n*** starting test: ' + testname + ' **'))
 
-    def fail_save_one_genome(self, params, error, exception=ValueError, contains=False):
+    def fail_save_genome(self, params, error, exception=ValueError, contains=False, mass=False):
         with self.assertRaises(exception) as context:
-            self.getImpl().save_one_genome(self.ctx, params)
+            if mass:
+                self.genome_interface.save_genome_mass(params)
+            else:
+                self.getImpl().save_one_genome(self.ctx, params)
         if contains:
             self.assertIn(error, str(context.exception))
         else:
             self.assertEqual(error, str(context.exception))
 
-    def check_save_one_genome_output(self, ret, genome_name):
+    def check_save_one_genome_output(
+        self,
+        ret,
+        genome_name,
+        data_type=_KBASE_GENOME,
+        warnings=_GENOME_FILE_WARNINGS
+    ):
         self.assertTrue('info' in ret)
+        self.assertTrue('warnings' in ret)
 
+        # Check info
         genome_info = ret['info']
         self.assertEqual(genome_info[1], genome_name)
-        self.assertEqual(genome_info[2].split('-')[0], 'KBaseGenomes.Genome')
+        self.assertEqual(genome_info[2].split('-')[0], data_type)
+        self.assertTrue(datetime.strptime(genome_info[3], '%Y-%m-%dT%H:%M:%S+%f'))
         self.assertEqual(genome_info[5], self.user_id)
+        self.assertEqual(genome_info[6], self.wsID)
+        self.assertEqual(genome_info[7], self.wsName)
+
+        # Check warnings
+        self.assertEqual(ret['warnings'], warnings)
 
     def test_bad_one_genome_params(self):
         self.start_test()
         invalidate_params = {'missing_workspace': 'workspace',
                              'name': 'name',
                              'data': 'data'}
-        error_msg = '"workspace" parameter is required, but missing'
-        self.fail_save_one_genome(invalidate_params, error_msg)
+        error_msg = "Exactly one of a 'workspace_id' or a 'workspace' parameter must be provided"
+        self.fail_save_genome(invalidate_params, error_msg)
```

Comment:
Did we update the spec for save_one_genome to add workspace Id as a param?

Reply:
Nope. Do you think I should add workspace_id to the spec? I am inclined to keep the original API intact. The `workspace_id` parameter is introduced in the non-public method save_genome_mass.

Reply:
It's backwards compatible right?

Reply:
added

Reply:
:+1: